### DISCUSSION TITLE
source-controller unable to locate s3 bucket

### DISCUSSION THREAD

**Discussion by ajayOO8** [fluxcd/flux2#3941] [Q&A]
I have recently migrated my k8s version from 1.21 to 1.22. Since then `bucket` is complaining the following

> failed to confirm existence of '< bucket >' bucket: Head "https://s3.dualstack.us-east-1.amazonaws.com/< bucket >/": context deadline exceeded

I tried to work out these checks https://fluxcd.io/flux/components/source/buckets/#failed-bucket  but they all seems to be working for me. I am  still not able to figure out the root cause for this issue.

debug stacktrace of source-controller:
> github.com/fluxcd/pkg/runtime/events.(*Recorder).Eventf 
    github.com/fluxcd/pkg/runtime@v0.13.2/events/recorder.go:113 
github.com/fluxcd/source-controller/internal/reconcile/summarize.RecordContextualError 
    github.com/fluxcd/source-controller/internal/reconcile/summarize/processor.go:47 
github.com/fluxcd/source-controller/internal/reconcile/summarize.(*Helper).SummarizeAndPatch 
    github.com/fluxcd/source-controller/internal/reconcile/summarize/summary.go:180 
github.com/fluxcd/source-controller/controllers.(*BucketReconciler).Reconcile.func1 
    github.com/fluxcd/source-controller/controllers/bucket_controller.go:278 
github.com/fluxcd/source-controller/controllers.(*BucketReconciler).Reconcile 
    github.com/fluxcd/source-controller/controllers/bucket_controller.go:305 
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile 
    sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:114 
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler 
    sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:311 
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem 
    sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:266 
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2 
    sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:227

is there some compatibility issue with k8s v1.22. I have other environments with v1.21 where same configurations seems to be working just fine.

environment:
k8s: 1.22
source-controller: v0.22

bucket manifest is following(i am using SA for IAM auth):
> apiVersion: v1
> items:
> - apiVersion: source.toolkit.fluxcd.io/v1beta2
>   kind: Bucket
>   metadata:
>     finalizers:
>     - finalizers.fluxcd.io
>     labels:
>       kustomize.toolkit.fluxcd.io/name: common-flux-system
>       kustomize.toolkit.fluxcd.io/namespace: flux-system
>     name: < bucket-name>
>     namespace: flux-system
>   spec:
>     bucketName: < bucket-name>
>     endpoint: s3.amazonaws.com
>     interval: 5m0s
>     provider: aws
>     region: us-east-1
>     timeout: 30s
> kind: List

**Comment by makkes** [comment]
Thanks for your feedback @ajayOO8. I see you're running source-controller v0.22 which is more than a year old. I would encourage you to upgrade to the latest version to increase the chances of pinning down the issue to its source.

"context deadline exceeded" suggests a network issue, though. Are you sure that the source-controller pod is able to properly connect to the S3 server?

**Reply by ajayOO8** [reply]
I tried to exec into the source-controller container and made a wget to the s3 bucket on the URL ("https://s3.dualstack.us-east-1.amazonaws.com/< bucket >/") suggested on the logs. it returned 403, which means it is indeed able to connect to the bucket.
